docs/fp_primer.md

```diff
@@ -0,0 +1,495 @@
+with precise meanings, which will be elaborated here. 
+
+Additionally, any specific notation used will be specified here to avoid
+confusion.
+
+### Operations
+
+The CTS tests for the proper execution of f32 builtins, i.e. sin, sqrt, abs,
+etc, and expressions, i.e. *, /, &&, etc. These collectively can be referred to
+as f32 operations.
+
+Operations, which can be thought of as mathematical functions, are mappings from
+a set of inputs to a set of outputs.
+
+Denoted `f(x, y) = X`, where f is a placeholder or the name of the operation,
+lower case variables are the inputs to the function, and uppercase variables are
+the outputs of the function.
+
+Operations have one or more inputs and an output. Being a f32 operation means
+that the primary space for input and output values is f32, but there is some
+flexibility in this definition. For example operations with values being
+restricted to a subset of integers that are representable as f32 are often
+referred to as being f32 based. 
+
+Values are generally floats, integers, booleans, vector, and matrices. Consult
+the WGSL spec for the exact list of types and their definitions.
+
+For complex outputs where there are multiple values being returned, there is a
+single result value made of structured data. Whereas inputs handle this by
+having multiple input parameters.
+
+Some examples of different types of operations:
+
+`multiplication(x, y) = X`, which represent `x * y`, takes in f32 values, `x`
+and `y`, and produces a f32 value `X`.
+
+`and(x, y) = X`, which represents `x && y`, again takes in f32 values, but in 
+this case returns a boolean value.
+
+`ldexp(x, y) = X`, which builds a f32 takes, takes in a f32 values `x` and a
+restricted integer `y`.
+
+### Domain, Range, and Intervals
+
+For an operation `f(x) = X`, the interval of valid values for the input, `x`, is
+called the domain, and the interval for valid results, `X`, is called the range.
+
+An interval, `[a, b]`, is a set of values that contains `a`, `b`, and all the 
+values between them.
+
+Open-ended intervals, i.e. ones that don't include `a` and/or `b`, are avoided,
+and are called out explicitly when they occur.
+
+The convention in this doc and the CTS code is that `a <= b`, so `a` can be
+referred to as the beginning of the interval and `b` as the end of the interval.
+
+When talking about intervals, this doc and the code endeavours to avoid using 
+the term **range** to refer to the span of values that an interval covers, 
+instead using the term bounds to avoid confusion of terminology around output of
+operations.
+
+## Accuracy
+
+As mentioned above floating point numbers are not able to represent all of the
+possible values over their bounds, but instead represent discrete values in that
+interval, and approximate the remainder.
+
+Additionally, floating point numbers are not evenly distributed over their in 
+interval, but instead are clustered closer together near zero, and further apart
+near the edges.
```

Comment:
"edges" is ambiguous?  Like,  they're bunched closer to the bottom edge of the interval [1000, 1000000].

I think you mean over the whole finite range of floats, so:

> ... are note evenly distributed over the real number line, but instead are ..  .. and further apart as their magnitudes grow.